modules/memberships/class-jetpack-memberships.php

```diff
@@ -277,5 +288,58 @@ public static function get_blog_id() {
 	public static function get_connected_account_id() {
 		return get_option( self::$connected_account_id_option_name );
 	}
+
+	/**
+	 * Whether Recurring Payments are enabled.
+	 *
+	 * @return bool
+	 */
+	public static function is_enabled_jetpack_recurring_payments() {
+		// For WPCOM sites.
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM && function_exists( 'has_any_blog_stickers' ) ) {
+			$site_id = get_current_blog_id();
+			return has_any_blog_stickers( array( 'personal-plan', 'premium-plan', 'business-plan', 'ecommerce-plan' ), $site_id );
+		}
+
+		// For Jetpack sites.
+
+		// FIXME: We enable Recurring Payments on all Jetpack sites, including those on a Free plan.
+		// The reason is that in Jetpack, we cannot yet support the 'new', Upgrade and Stripe Nudge-based
+		// flow, since the post-checkout redirect skips the Jetpack Onboarding Checklist in Calypso,
+		// where Akismet and VaultPress installation is triggered.
+		// Instead, we continue to show the legacy, in-block prompts to upgrade the site's plan, and to
+		// connect Stripe. This however means that we have to display the block also if the site's on a Free plan.
+		//
+		// Check p7rd6c-23C-p2 for details and progress. If ready, uncomment the right-hand side of the line with the
+		// return statement below.
```

Comment:
We should update this comment to reflect the change on L317, [see](https://github.com/Automattic/jetpack/pull/13394#discussion_r321672984)